### ISSUE TITLE
[Bug]: fully-sharded fused MoE W13 LoRA uses wrong slice offset after all-gather when local LoRA rank is 1

### ISSUE THREAD

**Issue by yiwenlu66** [vllm-project/vllm#42718]
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
I could not provide a useful collect_env.py output from this reporting machine:

1. The local checkout used for source inspection is a source-only checkout and does not have vLLM installed as a package.
2. The runtime where the failure was observed is an internal multi-node deployment; its API host is not the GPU worker runtime and cannot import the worker-side vLLM dependencies directly.

The source-level issue below is present on vllm-project/vllm main at:
75fd68c7a57a2f9560919a06863ea739ab46554c
```

</details>

### Describe the bug

`fully_sharded_loras=True` with fused MoE LoRA can compute the wrong W13 slice in the expand kernel when the local LoRA rank per TP rank is 1.

This is not the same as #32235. That issue was about the fused MoE LoRA launch grid size. This issue is about the post-all-gather memory layout of the W13 intermediate and the slice offset used by the expand kernel.

Affected source path on current `main`:

1. `FusedMoEWithLoRA` uses two W13 slices for gated MoE:
https://github.com/vllm-project/vllm/blob/75fd68c7a57a2f9560919a06863ea739ab46554c/vllm/lora/layers/fused_moe.py#L37-L40

2. `PunicaWrapperGPU.add_lora_w13` passes both slices into one combined fused MoE LoRA call:
https://github.com/vllm-project/vllm/blob/75fd68c7a57a2f9560919a06863ea739ab46554c/vllm/lora/punica_wrapper/punica_gpu.py#L597-L612

3. In fully-sharded mode, the intermediate rank dimension is all-gathered before expand:
https://github.com/vllm-project/vllm/blob/75fd68c7a57a2f9560919a06863ea739ab46554c/vllm/lora/ops/triton_ops/fused_moe_lora_op.py#L814-L827

4. `_fused_moe_lora_expand` flattens the tensor with `.view(-1, lastdim)` and passes `slice_a_size=a_intermediate_cache1.numel() // num_slices`:
https://github.com/vllm-project/vllm/blob/75fd68c7a57a2f9560919a06863ea739ab46554c/vllm/lora/ops/triton_ops/fused_moe_lora_op.py#L581-L657

5. The kernel computes the slice base as `cur_a_ptr = a_ptr + slice_id * slice_a_size`:
https://github.com/vllm-project/vllm/blob/75fd68c7a57a2f9560919a06863ea739ab46554c/vllm/lora/ops/triton_ops/fused_moe_lora_op.py#L295

That assumes each W13 slice is stored as one contiguous block after all-gather. This assumption is false for local rank 1.

Minimal stride reproducer:

```python
import torch

world = 16
num_slices = 2
tokens = 3
topk = 1
local_rank = 1

x = torch.empty(num_slices, tokens, topk, local_rank)
input_size = tuple(x.shape)

out = torch.empty((world,) + input_size)
z = out.reshape((world,) + input_size).movedim(0, x.dim() - 1)
z = z.reshape(input_size[: x.dim() - 1] + (input_size[-1] * world,))

print("shape:", z.shape)
print("stride:", z.stride())
print("actual slice1 offset:", z.stride(0))
print("kernel-assumed slice1 offset:", z.numel() // num_slices)
```

Observed output:

```text
shape: torch.Size([2, 3, 1, 16])
stride: (3, 1, 96, 6)
actual slice1 offset: 3
kernel-assumed slice1 offset: 48
```

So for `slice_id=1`, the kernel reads from `base + 48`, but the correct W3 slice starts at `base + 3`.

For local rank greater than 1, PyTorch materializes a contiguous copy during the final reshape, so the same assumption can accidentally hold. Example with `local_rank=4`:

```text
shape: torch.Size([2, 3, 1, 64])
stride: (192, 64, 64, 1)
actual slice1 offset: 192
kernel-assumed slice1 offset: 192
```

Expected behavior:

The expand kernel should address W13 slices using the actual tensor stride/layout after all-gather, or the caller should avoid the combined two-slice path for fully-sharded W13.

One safe implementation pattern is to run each W13 slice as a separate `num_slices=1` expand call in fully-sharded mode. That avoids the invalid `slice_id * (numel // num_slices)` offset entirely.

Observed runtime symptom:

In a TP=16, rank=16 fully-sharded fused MoE LoRA setup, the rank per TP shard is 1. The routed FusedMoE LoRA path produced non-finite first-token logits. A kernel discriminator localized the problem to combined W13; W2 and single-slice W13 paths were finite, and splitting W13 into separate one-slice expand calls removed the local-rank-1 corruption.

The FP8 fused MoE LoRA path appears to repeat the same pattern:
https://github.com/vllm-project/vllm/blob/75fd68c7a57a2f9560919a06863ea739ab46554c/vllm/lora/ops/triton_ops/fused_moe_lora_fp8_op.py#L620-L668

### Before submitting a new issue...

- [x] I searched existing issues. The closest issue I found was #32235, but it covers a different fused MoE LoRA bug.

**Comment by abinggo** [comment]
I'd like to take this. The stride reproducer makes the failure mode crystal clear (`stride(0)=3` vs `numel()//2=48`). Will follow your suggested direction — split each W13 slice into its own `num_slices=1` expand call on the fully-sharded path — and send a draft PR with the stride reproducer rolled into a regression test.
